src/main/scala/nl.knaw.dans.easy.deposit/docs/DatasetXml.scala

```diff
@@ -0,0 +1,172 @@
+        { <ddm:accessRights>{ dm.getAccessRights.category.toString }</ddm:accessRights> }
+      </ddm:profile>;
+
+    <ddm:DDM
+      xmlns:dc="http://purl.org/dc/elements/1.1/"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:dcterms="http://purl.org/dc/terms/"
+      xmlns:dcx-dai="http://easy.dans.knaw.nl/schemas/dcx/dai/"
+      xmlns:ddm="http://easy.dans.knaw.nl/schemas/md/ddm/"
+      xsi:schemaLocation="http://easy.dans.knaw.nl/schemas/md/ddm/ http://easy.dans.knaw.nl/schemas/md/2017/09/ddm.xsd"
+    >{ ddmProfile }
+      <ddm:dcmiMetadata>
+        { dm.alternativeTitles.getNonEmpty.map(str => <dcterms:alternative>{ str }</dcterms:alternative>).addAttr(lang) }
+        { dm.getContributors.map(author => <dcx-dai:contributorDetails>{ authorDetails(author) }</dcx-dai:contributorDetails>) }
+        { dm.getRightsHolders.map(author => <dcterms:rightsHolder>{ author.toString }</dcterms:rightsHolder>) }
+        { dm.publishers.getNonEmpty.map(str => <dcterms:publisher>{ str }</dcterms:publisher>).addAttr(lang) }
+        { dm.sources.getNonEmpty.map(str => <dc:source>{ str }</dc:source>).addAttr(lang) }
+        { (dm.getOtherDates.filter(_.hasScheme) :+ dateSubmitted).map(date => <x xsi:type={date.scheme.getOrElse("")}>{ date.value }</x>.withLabel(date.qualifier.toString)) }
+        { dm.getOtherDates.filterNot(_.hasScheme).map(date => <x>{ date.value }</x>.withLabel(date.qualifier.toString)) }
+        { dm.license.getNonEmpty.map(str => <dcterms:license>{ str }</dcterms:license>) /* xsi:type="dcterms:URI" not supported by json */ }
+      </ddm:dcmiMetadata>
+    </ddm:DDM>
+  }
+
+  private def authorDetails(author: Author)
+                           (implicit lang: Option[Attribute]) = {
+    if (author.surname.isEmpty)
+      author.organization.toSeq.map(orgDetails(author.role))
+    else
+      <dcx-dai:author>
+        { author.titles.getNonEmpty.map(str => <dcx-dai:titles>{ str }</dcx-dai:titles>).addAttr(lang) }
+        { author.initials.getNonEmpty.map(str => <dcx-dai:initials>{ str }</dcx-dai:initials>) }
+        { author.insertions.getNonEmpty.map(str => <dcx-dai:insertions>{ str }</dcx-dai:insertions>) }
+        { author.surname.getNonEmpty.map(str => <dcx-dai:surname>{ str }</dcx-dai:surname>) }
+        { author.role.toSeq.map(role => <dcx-dai:role>{ role.key }</dcx-dai:role>) }
+        { author.organization.getNonEmpty.map(orgDetails()) }
+      </dcx-dai:author>
+  }
+
+  private def orgDetails(role: Option[SchemedKeyValue[String]] = None)
+                        (organization: String)
+                        (implicit lang: Option[Attribute]) =
+      <dcx-dai:organization>
+        { role.toSeq.map(role => <dcx-dai:role>{ role.key }</dcx-dai:role>) }
+        { <dcx-dai:name>{ organization }</dcx-dai:name>.addAttr(lang) }
+      </dcx-dai:organization>
+
+  private implicit class SubmittedDatasetMetadata(dm: DatasetMetadata) extends DatasetMetadata {
```

Comment:
The recalculations preparing for the conversion started with some simple inline calls to generic implicit methods like the current getNonEmpty, but the number of lines I needed to extract into a value grew and grew. I wanted to clean up the XML conversion method by grouping the growing number of values that needed recalculation. 

We need a case class during the draft stage for json conversions so we can't extend the class for the submit stage. Adding many submit values to the base class would cause needles calculations during the draft stage, but most of all we should not apply submit requirements during the draft stage. At that point the idea for a complex implicit class was born.

For now I could reduce the original clutter for the apply method into rightsHolders and SubmittedDates. Don't yet know what will emerge in the next implementation phase with temporal/spatial coverage and so on.